src/App.vue

```diff
@@ -1,81 +1,32 @@
 <script setup>
 import IpsetInput from "./components/IpsetInput.vue";
-import { reactive, ref, watch } from "vue";
+import { computed, ref, watch, watchEffect } from "vue";
 
-const refs = ref([]);
+function isValidateIP(value) {
+  return /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/.test(value);
+}
 
-function Validate(val) {
-  if (isNaN(Number(val)) || Number(val) > 255 || val === "") return true;
+function toFormData(form) {
+  return Object.fromEntries(new FormData(form).entries());
 }
 
 const isError = ref(false);
 
-function Submit() {
-  isError.value = false;
+function onSubmit(e) {
+  const ip = Array.from({ ...toFormData(e.target), length: 4 }).join(".");
 
-  refs.value.map((item) => {
-    if (Validate(item.dom.value)) {
-      isError.value = true;
-    }
-  });
+  isError.value = !isValidateIP(ip);
 }
 
-const ip = reactive({
-  ip1: "",
-  ip2: "",
-  ip3: "",
-  ip4: "",
-});
-
-watch(
-  () => ip.ip1,
-  () => {
-    if (ip.ip1.length >= 3) {
-      ip.ip2 = ip.ip1.substring(3);
-      ip.ip1 = ip.ip1.substring(0, 3);
-      refs.value[2].dom.focus();
-    }
-  }
-);
-
-watch(
-  () => ip.ip2,
-  () => {
-    if (ip.ip2.length >= 3) {
-      ip.ip3 = ip.ip2.substring(3);
-      ip.ip2 = ip.ip2.substring(0, 3);
-      refs.value[3].dom.focus();
-    }
-
-    if (ip.ip2.length === 0 && ip.ip1.length > 0) {
-      refs.value[1].dom.focus();
-    }
-  }
-);
+function onChange(e) {
+  const { value, maxLength, name } = e.target;
 
-watch(
-  () => ip.ip3,
-  () => {
-    if (ip.ip3.length >= 3) {
-      ip.ip4 = ip.ip3.substring(3);
-      ip.ip3 = ip.ip3.substring(0, 3);
-      refs.value[4].dom.focus();
-    }
+  if (value.length !== maxLength) return;
 
-    if (ip.ip3.length === 0 && ip.ip2.length > 0) {
-      refs.value[2].dom.focus();
-    }
-  }
-);
-
-watch(
-  () => ip.ip4,
-  () => {
-    if (ip.ip4.length === 0 && ip.ip3.length > 0) {
-      refs.value[3].dom.focus();
-    }
-  }
-);
+  const form = e.target.closest("form");
+  const next = Number(name) + 1;
+  form.querySelector(`input[name="${next}"]`)?.focus();
+}
```

Comment:
我有個大膽的想法讓他按back到空白後會退回前一個input
```suggestion
   if (value.length !== maxLength && value.length !== 0) return;

  const form = e.target.closest("form");
  const current = Number(name);

  if (value.length === 0) {
    if (current - 1 >= 0)
      form.querySelector(`input[name="${current - 1}"]`)?.focus();
    return;
  }
  form.querySelector(`input[name="${current + 1}"]`)?.focus();
}
```